### ISSUE TITLE
Continuing training from loaded model

### ISSUE THREAD

**Issue by Schiboni** [keras-team/keras#11077]
Hi,
despite the fact that other discussions can be found on the topic, i still cannot find a solution to it.

I am training a network for one epoch and saving it. I would like to load the network and to continue training my model. 
In order to do that i use the commands:
model.save()
model.load()
When i start training at epoch 0 my loss is around 4.x and the accuracy around 0.19. When i finish the first epoch the loss is around 0.2 and accuracy around 94.x.
When i continue training after loading the model my loss is around 7.x and the accuracy 30.x.

Is anything alright? Should the loss be way lower?

**Comment by adityabalu** [comment]
Could you give more details? what is the model?

**Comment by Schiboni** [comment]
Thanks for your answer.
The model is the same of the paper at the link below, Figure 3.
http://www.mdpi.com/1424-8220/16/1/115/htm
There is the input layer, four convolutional layers and two dense layers (lstm). As the paper, i am processing inertial sensor data.

**Comment by Schiboni** [comment]
epoch = 0
    model_name = get_model_path(winSize, step_null, subjects, NUM_CONV_LAYS, epoch, step_class, learning_rate0)
    while os.path.isfile(save_dir + 'model/' + model_name + '.h5'):
        epoch += 1
        model_name = get_model_path(winSize, step_null, subjects, NUM_CONV_LAYS, epoch, step_class, learning_rate0)
    epoch -= 1
    if epoch == -1:
        raise('No pretrained model was found')
    prev_model_name = get_model_path(winSize, step_null, subjects, NUM_CONV_LAYS, epoch, step_class, learning_rate0)
    epoch +=1

    model = load_model(save_dir + 'model/' + prev_model_name +'.h5')

numChunkTot = chunks_nullclass * NUM_CLASSES
inst_todiscard = (numChunkTot) % (batchSize * ngpus)
 inst_toprocess = int(numChunkTot - inst_todiscard)
num_batches = int((inst_toprocess)/(batchSize * ngpus))


if continuing == 0:
    epoch = 0

generator_data = gen.generator_batches(batchSize, subjects, NUM_CLASSES, data_column_names,
                          gestures_label, position, winSize, inst_toprocess, root, data_type, nfiles, step_class, 
                           NB_SENSOR_CHANNELS, oversampling_fac, step_null)

model_name = get_model_path(winSize, step_null, subjects, NUM_CONV_LAYS, epoch, step_class, learning_rate0)
K.set_value(model.optimizer.lr, genmod.get_decayed_rate(decay_rate, epoch, learning_rate0))

t0 = time.time()

H = model.fit_generator(generator=generator_data, epochs=1, steps_per_epoch=num_batches, verbose=1, use_multiprocessing=True, workers=2)

t1 = time.time()
print('Time per epoch ', t1)
# Save model and weights
model.save(save_dir + 'model/' + model_name+'.h5')
# save history
tst.safe_obj_w(H.history, save_dir + 'history/' + model_name)
`

**Comment by Schiboni** [comment]
Please not that i fix the learning rate with decay 0 when compiling, and after each epoch i change the learning rate with a decay rate of 0.9, using K.set_value.

**Comment by Schiboni** [comment]
@gabrieldemarmiesse regardless you spot a mistake or not, may you enlist those many little things you mentioned? So i can have a clear idea. Many thanks in advance.

**Comment by gabrieldemarmiesse** [comment]
Sure. Here are the common mistakes and gotchas I'm used to see:

* The generator loading the data can have a strange behavior, for example, the training and validation step are not the same as the ones that your generator is actually using.
* Keras does a running mean to compute the losses and metrics. So the order of the data matters at training time when the metrics are displayed with the progress bar.
* optimizers can behave in a weird way when save (ex: you reduce the learning rate manually, and then it's reset when you load the model, if you use a callback).

**Comment by Schiboni** [comment]
I do not have idea how to proceed. I have been stuck for at least one month on this problem. How shall i fix this problem?

**Comment by Schiboni** [comment]
Thanks i will try that. Another quick question: i read somewhere that keras change the learning rate, as default, during an epoch, is it true?

**Comment by gabrieldemarmiesse** [comment]
If the decay is 0, I don't think keras does any unexpected stuff like changing the learning rate. Keras bases it's optimizer's implementation on the original papers.

**Comment by Schiboni** [comment]
OK, another question: when i resume the training do i need to compile again or not?

**Comment by Schiboni** [comment]
Hi,
i noticed an important thing. 

H = model.fit_generator(generator=generator_data, epochs=1, steps_per_epoch=num_batches, verbose=1, use_multiprocessing=True, workers=2)
H = model.fit_generator(generator=generator_data, initial_epoch=1, epochs=10,  steps_per_epoch=num_batches, verbose=1, use_multiprocessing=True, workers=2)

If i run this code, i.e., i do not save and load the model, i just continue to train, the problem remains, i.e, the accuracy drops down. I wonder if that is not natural then, and i do not have to worry about it.
